### PR TITLE
Theme Discovery: Add see all link in ThemeCollection

### PR DESCRIPTION
**Pull request by BogdanUngureanu**
Related to https://github.com/Automattic/dotcom-forge/issues/3959
## Proposed Changes

* Add a See all link in the Theme Collection

## Testing Instructions

* Use the Calypso live link and go to /themes in incognito
* Enable the feature flag: flags=themes/discovery-lots
* Notice a See all link on the page
* Click on the Premium one and on the Partner one
* Notice that you are redirected to /themes/premium and /themes/marketplace

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by BogdanUngureanu** [comment]
It still needs a bit of styling. Richard, I know how much you like CSS, so I'm assigning it to you :D

**Comment by rcrdortiz** [comment]
> It still needs a bit of styling. Richard, I know how much you like CSS, so I'm assigning it to you :D

A bit more than styling in the end 😆.

This is now ready for review.

File: client/components/theme-collection/index.tsx [hunk 1]
```diff
@@ -83,6 +88,9 @@ export default function ThemeCollection( {
 					<div className="theme-collection__description">{ description }</div>
 				</div>
 				<div className="theme-collection__carousel-controls">
+					<Button className="theme-collection__see-all" onClick={ onSeeAll }>
```
Comment (BogdanUngureanu) [inline client/components/theme-collection/index.tsx:91]: I wonder why a regular link won't work. 🤔

**Comment by BogdanUngureanu** [comment]
@rcrdortiz Looks good! However, one small thing I've noticed: When you click on the `see all` from the Partner themes collection, the page keeps its scrolling position. I think it should be pointed at the top of the page.

File: client/components/theme-collection/index.tsx (hunk 1 above)
Comment (rcrdortiz) [reply client/components/theme-collection/index.tsx:91]: A plain link does work, but we're going to want to track these clicks. When I work on the tracking issue this implementation might change depending on what I find during the implementation.

**Comment by rcrdortiz** [comment]
> @rcrdortiz Looks good! However, one small thing I've noticed: When you click on the `see all` from the Partner themes collection, the page keeps its scrolling position. I think it should be pointed at the top of the page.

The function that scrolls the Theme Showcase wasn't working for the LoTS. Since it wasn't too much of a change I did it in this PR instead of in a follow-up.

**Comment by BogdanUngureanu** [comment]
Works as expected! 🎉  We can ship it! :) 

One thing (unrelated to this PR and that can be handled in a separate PR): We don't have a way for the user to go back to the collections page.

For example, when they click on the see all link for a collection (e.g. partner), clicking on the "Recommended" tab again will just load the partner themes. For them to go back, they'll have to change the tier in the search bar, which is not that obvious.


Btw, I can't approve it because I'm the author of the PR. 🤣

**Review by Copons — APPROVED** [review]
LGTM!

I have a few non-blocking notes.

> One thing (unrelated to this PR and that can be handled in a separate PR): We don't have a way for the user to go back to the collections page.
>
> For example, when they click on the see all link for a collection (e.g. partner), clicking on the "Recommended" tab again will just load the partner themes. For them to go back, they'll have to change the tier in the search bar, which is not that obvious.

This is very important, and I don't have a good answer. We can chat about it tomorrow, I'm curious to hear your opinions. 🙂 

Screenshot 2023-10-11 at 16 43 10

There is a weird misalignment between the title/buttons element and the carousel. Those spacings also vary responsively.
Thumbnails should be aligned with the title on the left and the arrow on the right.

**Pull request merged**